src/Features/Core/Portable/ExtractMethod/MethodExtractor.AnalyzerResult.cs

```diff
@@ -162,6 +140,40 @@ public IEnumerable<VariableInfo> GetVariablesToSplitOrMoveOutToCallSite(Cancella
                 return _variables.Where(v => v.GetDeclarationBehavior(cancellationToken) is DeclarationBehavior.SplitOut or
                                                  DeclarationBehavior.MoveOut);
             }
+
+            public VariableInfo GetOutermostVariableToMoveIntoMethodDefinition(CancellationToken cancellationToken)
+            {
+                using var _ = ArrayBuilder<VariableInfo>.GetInstance(out var variables);
+                variables.AddRange(this.GetVariablesToMoveIntoMethodDefinition(cancellationToken));
+                if (variables.Count <= 0)
+                    return null;
+
+                VariableInfo.SortVariables(variables);
+                return variables[0];
+            }
```

Comment:
this is a move.  but it allows us to know and check where an extract local function needs to go before doing any transformations.